### DISCUSSION TITLE
Meaning of CPU graph

### DISCUSSION THREAD

**Discussion by j-wezel** [aristocratos/bpytop#244] [Q&A] [answered]
I don't understand the big CPU graph on the left. What is the meaning of the upward scale vs the downward scale?

**Comment by Rohit544** [comment]
The CPU graph shows the percentage ofCPU processing capacity being used against operations on the array. Even when no external I/O is occurring, the CPUutilization percentage can be non-zero because the storage operating system might be doing background operations and monitoring.

**Reply by j-wezel** [reply]
Thanks. But I still do not understand what the distinction is between the part from the middle line upwards and the part from the middle line downwards. They are not the same, hence must depict two different aspects. But what are these two aspects?

**Comment by aristocratos** [comment, marked as answer]
The big CPU graph shows total usage for the complete CPU package. It consists of two graphs, one normal and one inverted graph to give the appearance of growing out from the middle. 
They both are generated using the same data, but depending on the size of the CPU box (even or uneven number of lines), the middle point might be off.
Since if the size of the box is uneven, one of the graphs will be 1 line taller than the other.
For the moment the only way to rectify this is to resize your terminal until they are the same size.
(There are plans to implement more CPU graph modes like 1 single graph or bigger graphs for core usage split evenly to fit the box, etc)

**Reply by j-wezel** [reply]
Thanks, that is clear now. As a suggestion, one could use the lower part to show system time, or nice time.

**Reply by aristocratos** [reply]
> As a suggestion, one could use the lower part to show system time, or nice time.

That's a good idea, but I think I can do even better. Add options in the menu for choosing between (user, system, idle, nice, iowait, irq, etc...) for each graph :)

**Reply by dnhkng** [reply]
Was this ever implemented?
